addons/mail/static/tests/core/record.test.js

```diff
@@ -1054,26 +1054,29 @@ test("insert with id relation keeps existing field values", async () => {
     expect(member2.is_internal).toBe(true);
 });
 
-test("Re-assign on Many relation index with a One inverse should delete record from record list", async () => {
-    // Assign on record list index deletes the old record from record list.
-    // A typo can easily happen in internal code of JS and instead of calling
-    // recordList.delete() it calls oldRecord.delete()!
+test("Can assign new record on Many field with One inverse", async () => {
     (class Thread extends Record {
         static id = "name";
         name;
         files = Record.many("File", { inverse: "thread" });
     }).register(localRegistry);
     (class File extends Record {
         static id = "name";
+        name;
         thread = Record.one("Thread", { inverse: "files" });
     }).register(localRegistry);
     const store = await start();
-    store.Thread.insert("general");
-    store.File.insert(["file1.txt", "file2.txt"]);
-    store.Thread.get("general").files = ["file1.txt"];
-    store.Thread.get("general").files[0] = "file2.txt";
-    // Note: using .get() to check no record is deleted by mistake
-    expect(store.Thread.get("general").files.length).toBe(1);
-    expectRecord(store.Thread.get("general").files[0]).toEqual(store.File.get("file2.txt"));
-    expect(store.File.get("file1.txt").exists()).toBe(true);
+    const thread = store.Thread.insert("general");
+    const file1 = store.File.insert("file1.txt");
+    const file2 = store.File.insert("file2.txt");
+    thread.files.push(file1);
+    expect(thread.files.length).toBe(1);
+    expectRecord(thread.files[0]).toEqual(file1);
+    expectRecord(file1.thread).toEqual(thread);
+    expect(file2.thread).toBe(undefined);
+    thread.files[0] = file2;
+    expect(thread.files.length).toBe(1);
+    expectRecord(thread.files[0]).toEqual(file2);
+    expectRecord(file2.thread).toEqual(thread);
+    expect(file1.thread).toBe(undefined);
```

Comment:
No need to `.get()` because accessing relation in both ways implicitly checks exists on all records.